docs/index.md

```diff
@@ -0,0 +1,534 @@
+If you create a model that inherits from `BaseSettings`, the model initialiser will attempt to determine
+the values of any fields not passed as keyword arguments by reading from the environment. (Default values
+will still be used if the matching environment variable is not set.)
+
+This makes it easy to:
+
+* Create a clearly-defined, type-hinted application configuration class
+* Automatically read modifications to the configuration from environment variables
+* Manually override specific settings in the initialiser where desired (e.g. in unit tests)
+
+For example:
+
+```py
+from typing import Any, Callable, Set
+
+from pydantic import (
+    AliasChoices,
+    BaseModel,
+    ConfigDict,
+    ImportString,
+    RedisDsn,
+    PostgresDsn,
+    AmqpDsn,
+    Field,
+)
+from pydantic_settings import BaseSettings
+
+
+class SubModel(BaseModel):
+    foo: str = 'bar'
+    apple: int = 1
+
+
+class Settings(BaseSettings):
+    auth_key: str = Field(validation_alias='my_auth_key')
+    api_key: str = Field(validation_alias='my_api_key')
+
+    redis_dsn: RedisDsn = Field(
+        'redis://user:pass@localhost:6379/1',
+        validation_alias=AliasChoices('service_redis_dsn', 'redis_url')
+    )
+    pg_dsn: PostgresDsn = 'postgres://user:pass@localhost:5432/foobar'
+    amqp_dsn: AmqpDsn = 'amqp://user:pass@localhost:5672/'
+
+    special_function: ImportString[Callable[[Any], Any]] = 'math.cos'
+
+    # to override domains:
+    # export my_prefix_domains='["foo.com", "bar.com"]'
+    domains: Set[str] = set()
+
+    # to override more_settings:
+    # export my_prefix_more_settings='{"foo": "x", "apple": 1}'
+    more_settings: SubModel = SubModel()
+
+    model_config = ConfigDict(env_prefix = 'my_prefix_')  # defaults to no prefix, i.e. ""
+
+print(Settings().model_dump())
+"""
+{
+    'auth_key': 'xxx',
+    'api_key': 'xxx',
+    'redis_dsn': Url('redis://user:pass@localhost:6379/1'),
+    'pg_dsn': Url('postgres://user:pass@localhost:5432/foobar'),
+    'amqp_dsn': Url('amqp://user:pass@localhost:5672/'),
+    'special_function': <built-in function cos>,
+    'domains': set(),
+    'more_settings': {'foo': 'bar', 'apple': 1},
+}
+"""
+```
+
+<!--- {!.tmp_examples/settings_main.md!} --->
```

Comment:
As you can see the example code is injected directly here in the md file and this line is commented. Whenever we decided about the docs, we can uncomment this line and remove the above example code